src/tools/build-manifest/README.md

```diff
@@ -1,27 +1,34 @@
 # build-manifest
 
-This tool generates the manifests uploaded to static.rust-lang.org and used by
-rustup. The tool is invoked by the bootstrap tool.
+This tool generates the manifests uploaded to static.rust-lang.org and used by rustup.
+You can see a full list of all manifests at <https://static.rust-lang.org/manifests.txt>.
+
+This gets called by `promote-release` <https://github.com/rust-lang/promote-release> via `x.py dist hash-and-sign`.
+
+## Adding a new component
+
+There are several steps involved here.
+1. Add a new `Step` to `dist.rs`. This should usually be named after the filename of the uploaded tarball. See https://github.com/rust-lang/rust/pull/101799/files#diff-2c56335faa24486df09ba392d8900c57e2fac4633e1f7038469bcf9ed3feb871 for an example.
+    a. If appropriate, call `tarball.is_preview(true)` for the component.
+2. Add the component to the `Extended` tarball in `dist.rs`. This step cannot be skipped; without it, the artifacts will be uploaded and the component metadata will be present in the manifest, but rustup will not realize it's available.
```

Comment:
This feels pretty iffy. Do we really care about the extended tarball in rustup somehow? I don't think we put all components in there today, and I don't know that we want to even if we do.

Can you help me identify where rustup is consuming the extended tarball?

Reply:
I think I misidentified the problem here; I was missing the new component in the "extensions" section of the `rust` package, which is unrelated to the `Extended` tarball. https://github.com/rust-lang/rust/pull/102241/files#diff-1cb8c241c2567e1ac5beb8021e009f2ab69fdbaed8df0f4787cb87e268399112R446